### PR TITLE
Add support for stubbing undefined constants

### PR DESCRIPTION
**Pull request by sometimesfood**
This pull request adds support for stubbing undefined constants.

Typically, when I am stubbing a constant, I do so in order to ensure I know the current value in the context of a test. I do not usually care whether the constant is currently defined or not.

Having to add conditionals to my tests that check whether a constant is currently defined before stubbing it seems too cumbersome for me; hence this pull request.

### CODE REVIEW

File: lib/minitest/stub_const.rb [hunk 1]
```diff
@@ -37,15 +36,21 @@ def stub_const(name, val, &block)
...
+    if const_defined?(name)
+      begin
+        orig = const_get(name)
+        remove_const(name)
+        yield
+      ensure
+        const_set(name, orig)
+      end
+    else
+      yield
```
Comment (adammck) [inline lib/minitest/stub_const.rb]: Given that this method is named `stub_remove_const` rather than `ensure_const_stubbed`, I think that we should raise if the given const is undefined, rather than silently yield. Do you have a specific use-case for this in mind?

Comment (sometimesfood) [inline lib/minitest/stub_const.rb]: I see your point; I am not quite sure if this is the best way to handle this type of situation myself.

As to the use case: This method was the reason I submitted the pull request. I have a project where I have a soft dependency on mustache and use pencil_mustache as a fallback. In order to be able to test the fallback, I `stub_remove_const` the `Mustache` constant. However, currently this only works if the mustache gem is actually installed. (To make matters worse, Mustache does not support Ruby <2.0 anymore.)

Basically this change would make it easier to drop development dependencies that are only included for the sake of some constant being available at the test's runtime without having to duplicate every `stub_remove_const` block because of `const_defined?` checks.

Comment (adammck) [inline lib/minitest/stub_const.rb]: No, I think this makes sense. Thanks for elaborating. Would you mind expanding the docstring to clarify that the method does nothing (rather than e.g. raising) if the constant is undefined?

Comment (sometimesfood) [inline lib/minitest/stub_const.rb]: Great to hear that! Sure, I will amend my commit tomorrow.

Comment (sometimesfood) [inline lib/minitest/stub_const.rb]: Just pushed updated commits.

**Comment by adammck** [comment]
Great, thanks! Looks good to me.
